tools/internal/shell/interface.go

```diff
@@ -0,0 +1,62 @@
+package shell
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+//
+// Runner is an interface for running shell commands. It exists to
+// support mocking shell commands in unit tests.
+//
+// https://joshrendek.com/2014/06/go-lang-mocking-exec-dot-command-using-interfaces/
+//
+type Runner interface {
+	Run(cmd Command) error
+}
+
+// Error represents an error encountered running a shell command
+type Error struct {
```

Comment:
can we call this commandError or something more specific? This got me mixed up a bit thinking it was `error`

Reply:
I think this is actually conventional Golang, eg. zerolog has a `log.Error`, http has `http.Error`, exec has `exec.Error`, etc.

I actually tried calling it `ShellError` first but Goland kept screaming at me to not prefix my types with the package name, so :shrug: haha

Reply:
golaaaaaaaaang 🤷

Reply:
My personal take on this is that this is a standard go convention (not repeating the package name in your type) but kind of a silly one 🤷🏻 . The names look more intuitive when used from another package ie `shell.Error`. But within the package code itself it leads to ambiguous looking names. It's kind of a 6 of one half a dozen of the other type of thing IDK.